R/util.R

```diff
@@ -320,14 +324,15 @@ is_valid_percent <- function(str_pct_vec) {
   # ^              # beginning of string
   # (              # either/or group
   #     -?         # optional minus sign
-  #     1?         # optional hundreds digit
+  #     [1-9]?     # optional thousands digit
+  #     [1-9]?     # optional hundreds digit
   #     [0-9]{1,2} # some number 1 through 99
   #     %          # literal percent sign
   #   |            # OR
   #     —          # em dash
   # )              # close group
   # $              # end of string
-  cor_format <- grepl("^(-?1?[0-9]{1,2}%|—)$", str_pct_vec)
+  cor_format <- grepl("^(-?[1-9]?[1-9]?[0-9]{1,2}%|—)$", str_pct_vec)
```

Comment:
please-fix: this will fail for numbers in `1000:1099` because you're not allowing the hundreds digit to be zero.

If you want to match -9999 to 9999 I recommend:

````
^(-?([1-9][0-9]*|0)%|—)$
````
![Regular expression visualization](https://www.debuggex.com/i/38f_P0BfbpfYPJM8.png)

[Debuggex Demo](https://www.debuggex.com/r/38f_P0BfbpfYPJM8)